### DISCUSSION TITLE
Downloaded svg logo breaking the result

### DISCUSSION THREAD

**Discussion by oskay12** [chillerlan/php-qrcode#286] [Q&A] [answered]
for example i downloaded a svg from svgrepo. when i try to put it as logo its not working. Im opening the svg as text and delete some lines as comments. 
![donuts-cake-svgrepo-com](https://github.com/user-attachments/assets/3a541a85-cfb9-4c85-880b-88541a2eba3f)
this one i downloaded it and tried to put it to qr.

**Comment by oskay12** [comment]
![Ekran görüntüsü 2024-10-25 150006](https://github.com/user-attachments/assets/88f674b0-dcb0-453e-ab30-6de9db2cca1d)
This is the result.

**Reply by codemasher** [reply, marked as answer]
Ok, the error occurs when the SVG logo contains an XML header. The above image for example has this header:

```xml
<?xml version="1.0" ?><!-- Uploaded to: SVG Repo, www.svgrepo.com, Generator: SVG Repo Mixer Tools -->
```

When you remove it, the QR Code renders without issue.
However you will not yet see the logo, because the `<svg>` element has fixed `width` and `height` attributes that cause it to be positioned outside of the visible area. When you remove these too, the logo appears as expected.

```xml
<svg fill="#000000" width="800px" height="800px" viewBox="0 0 32 32" xmlns="http://www.w3.org/2000/svg" id="Layer_1" data-name="Layer 1">
```

You can see a bit of the logo here in the lower right corner: 
![image](https://github.com/user-attachments/assets/af70bf9e-7b90-461e-a791-cdea1b1a13f9)


With the unnecessary attributes removed it looks as follows:
```xml
<svg viewBox="0 0 32 32" xmlns="http://www.w3.org/2000/svg">
```

![svgWithLogoAndCustomShapes](https://github.com/user-attachments/assets/ebf9916e-b3bb-44d9-ab15-0192c8d0390d)

**Comment by oskay12** [comment]
Thank you very much

**Reply by codemasher** [reply]
Happy to help!
